buildpack.md

```diff
@@ -834,6 +834,10 @@ The buildpack ID:
 - MUST NOT be `config` or `app`.
 - MUST NOT be identical to any other buildpack ID when using a case-insensitive comparison.
 
+The buildpack version:
+- MUST be in the form `<X>.<Y>.<Z>` where `X`, `Y`, and `Z` are non-negative integers and must not contain leading zeros.
+   - Each element MUST increase numerically.
+   - Buildpack authors will to define how what changes will increment `X`, `Y`, and `Z`.
```

Comment:
```suggestion
   - Buildpack authors will define what changes will increment `X`, `Y`, and `Z`.
```

Reply:
This is actually more restrictive than semver, which allows things like `1.0.0-rc.1`. Should we loosen this up to include everything allowed by https://semver.org/, either by adding a regex or merely linking out?

Reply:
That's a good point - do we want to specifically say "use semver but the semantics are not enforced"? Or should I add another point about adding alphanumberic values?

Reply:
@ekcasey Does lifecycle parsesbuildpack versions? If it does, does it follow the semver spec?